### PR TITLE
frontend: resourceQuota: Add visual usage indicators for quota pressu…

### LINKED ISSUE #5873: frontend: resourceQuota: Add usage progress indicators to detail view
## Is your feature request related to a problem? Please describe the impact that the lack of the feature requested is creating.

Right now the `ResourceQuota` detail view shows quota usage as plain text (`Used` / `Hard` values in a table). It works, but when checking multiple resources or multiple namespaces I end up mentally doing the math every time to figure out how close something is to its limit.

For example, seeing `48` and `96` for CPU means stopping for a second to work out “okay, that's about half used”. It's small friction, but it adds up quickly when you're scanning quota pressure across namespaces during debugging or general cluster administration.

A visual indicator would make this much easier to read at a glance without changing the existing data or workflow.

## Describe the solution you'd like

Add a lightweight `LinearProgress` indicator alongside the existing `Used` / `Hard` values in the `ResourceQuota` details table.

The current table structure and text values would remain unchanged, the progress bar would just supplement the existing information so users can visually understand quota saturation without doing arithmetic in their head.

This could likely be implemented entirely in the frontend using the existing `resourceStats` data already exposed by the `ResourceQuota` class and MUI components already available in the project.

The existing utility functions in `lib/units.ts` also already seem sufficient for calculating ratios for CPU and memory resources.

Optionally, the same compact indicators could also be shown in the namespace detail quota section since that's often where quota pressure is checked first.

## What users will benefit from this feature?

Cluster admins and platform engineers working with namespaces that use `ResourceQuota`, especially in multi-tenant clusters where quota exhaustion is something you regularly need to keep an eye on.

## Are you able to implement this feature?

Yes , will follow up with a pr shortly.

## Additional context

While exploring the frontend codebase I noticed:

- `ResourceQuota` already exposes a `resourceStats` getter returning `{ name, hard, used }[]`
- the current detail view already renders these values in `frontend/src/components/resourceQuota/Details.tsx`
- `LinearProgress` from MUI is already available as a dependency

So this seems like a relatively self-contained frontend-only enhancement rather than a larger architectural change.

### PR DESCRIPTION
**Pull request by beep-boopp**
## Summary

Adds a "Usage" column to the ResourceQuota detail view table with a `LinearProgress` bar showing the used/hard ratio per resource. Color-coded: green under 80%, orange 80–90%, red 90%+.

## Related Issue

Fixes #5873

## Changes

- Added `getUsageRatio` helper, `QuotaUsageBar` component, and a 4th "Usage" column to `ResourceQuotaTable` in `Details.tsx`
- Added unit tests for `getUsageRatio` covering CPU, memory, count-type resources, and edge cases (12 tests, all passing)

## Steps to Test

1. Create a namespace with a ResourceQuota and deploy a workload with resource requests
2. Navigate to Configuration → Resource Quotas → click a quota
3. Detail view should show Resource / Used / Hard / Usage columns
4. Scale the workload to push usage past 80% and 90% to see color changes

## Screenshots

| Normal (< 80%) | Warning (80–90%) | Error (≥ 90%) |
|---|---|---|
| Normal usage | Warning usage | Error usage |

## Notes for the Reviewer

- No new dependencies — uses `LinearProgress` from MUI which is already in the project
- Parsing reuses existing `parseCpu` and `parseRam` from `lib/units.ts`
- Existing columns are untouched, this just adds a 4th column

### CODE REVIEW

File: frontend/src/components/resourceQuota/Details.test.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * limitations under the License.
+ */
+
+import { parseCpu, parseRam } from '../../lib/units';
+
+/**
+ * Mirror of getUsageRatio from Details.tsx — tested independently
+ * to avoid circular k8s module imports in the test environment.
+ */
+function getUsageRatio(name: string, used: string, hard: string): number {
```
Comment (sniok) [inline frontend/src/components/resourceQuota/Details.test.tsx]: this test isn't testing anything from the Details.tsx
they way this test is constructed doesn't make much sense since it doesn't test any implementation

if you're getting circular dependency issues you can try importing App [like this](https://github.com/kubernetes-sigs/headlamp/blob/main/frontend/src/components/resourceMap/graph/graphFiltering.test.ts#L25)

**Review by sniok — CHANGES_REQUESTED** [review]

File: frontend/src/components/resourceQuota/Details.test.tsx (hunk 1 above)
Comment (beep-boopp) [reply frontend/src/components/resourceQuota/Details.test.tsx]: Thanks for catching that, i was running into circular k8s module dependency issues when importing from Details.tsx directly, so i ended up duplicating the function as a workaround. Used the app pattern and have updated the test to import getUsageRatio directly now

File: frontend/src/components/resourceQuota/Details.tsx [hunk 2]
```diff
@@ -14,13 +14,73 @@
...
+      hardNum = parseRam(hard);
+      break;
+    default:
+      usedNum = parseInt(used, 10) || 0;
+      hardNum = parseInt(hard, 10) || 0;
+      break;
+  }
+
+  if (hardNum === 0) return 0;
+  return Math.min(usedNum / hardNum, 1);
```
[the remark this answers is not in the document]
Comment (beep-boopp) [reply frontend/src/components/resourceQuota/Details.tsx]: Updated to only cap the progress bar at 100% while showing the actual percentage in the text label.

File: frontend/src/components/resourceQuota/Details.tsx [hunk 3]
```diff
@@ -14,13 +14,73 @@
...
+      hardNum = parseCpu(hard);
+      break;
+    case 'memory':
+      usedNum = parseRam(used);
+      hardNum = parseRam(hard);
+      break;
+    default:
+      usedNum = parseInt(used, 10) || 0;
+      hardNum = parseInt(hard, 10) || 0;
+      break;
```
[the remark this answers is not in the document]
Comment (beep-boopp) [reply frontend/src/components/resourceQuota/Details.tsx:63]: Added storage and ephemeral storage cases to route through parseRam.

**Comment by beep-boopp** [comment]
@illume PTAL

**Comment by beep-boopp** [comment]
@illume PTAL all copilot comments have been resolved and factored in.

**Comment by beep-boopp** [comment]
@illume I have resolved the copilot comments and made the appropriate changes . PTAL.

**Comment by beep-boopp** [comment]
@illume PTAL

**Comment by beep-boopp** [comment]
@illume Sorry about the back and forth on these , import ordering was silly on my end. All Copilot comments addressed, PTAL. Should be good to go now

**Comment by beep-boopp** [comment]
@illume PTAL, did not realize we can't commit copilot suggestions directly

**Review by illume — APPROVED** [review]
🎉 thanks!
